### ISSUE TITLE
Connect an external app to consume my own api

### ISSUE THREAD

**Issue by NicolasLF** [api-platform/api-platform#2540]
Hello everyone,

I'm currently working on an api rest created with Symfony and ApiPlatform. This api is consumed by a front end created in Angular.
Users connect using JWT and everything is working perfectly.

Now, I need an external service provider to come and consume an endpoint from my api. I'd like to provide him with a client_key and a client_secret, or something like that, but that's where I get confused... 

It's not a user but an external app that has to consume a single endpoint of my api. What's the best way to authenticate it? How can I provide it with credentials without it having to request a token before each api call? 

In short, could you please tell me the right way to go about it?

**Comment by jmarsman** [comment]
Maybe use the [AccessTokenHandlerInterface](https://symfony.com/doc/current/security/access_token.html#using-the-access-token-authenticator)? 

Then add a `token` property to your User entity or add an extra table to provide more tokens. From there u can use this token to lookup the User for your external consumer.

From there you can add a role, let's say `ROLE_EXT_API` and make a firewall rule so only users in this role can access the endpoint you want.

**Comment by soyuka** [comment]
Closing per #2987. Questions are best asked on [GitHub Discussions](https://github.com/api-platform/core/discussions) or [Stack Overflow (api-platform tag)](https://stackoverflow.com/questions/tagged/api-platform.com). The framework now lives at `api-platform/core`.
